script.js

```diff
@@ -1,12 +1,78 @@
 //You can edit ALL of the code here
+let allEpisodes;
 function setup() {
-  const allEpisodes = getAllEpisodes();
+  allEpisodes = getAllEpisodes();
   makePageForEpisodes(allEpisodes);
 }
 
 function makePageForEpisodes(episodeList) {
   const rootElem = document.getElementById("root");
-  rootElem.textContent = `Got ${episodeList.length} episode(s)`;
+  rootElem.textContent = ""
+  const episodeCount= document.getElementById("episodeCount");
+  episodeCount.innerHTML= `Displaying ${episodeList.length}/73 episodes`;
+  //create select input
+  const selectEl=document.getElementById("episodeOptions");
+
+  episodeList.forEach(episode=>{
+    
+    //create body element
+    const divEl=document.createElement("div");
+    divEl.classList.add("container")
+    const episodeBtn=document.createElement("button");
+    episodeBtn.classList.add("episodeButton")
+    const episodeImg=document.createElement("img");
+    episodeImg.classList.add("episodeImg")
+    const episodeDescription=document.createElement("p");
+    episodeDescription.classList.add("episodeDescription")
+    //create option element
+    const optionEl=document.createElement("option");
+    optionEl.classList.add("optionElement")
+
+    // assign the text content
+    let epiNo= episode.number===10 ? episode.number : "0"+episode.number;
+    episodeBtn.innerHTML=`${episode.name} - S0${episode.season} E${epiNo}`;
+    episodeImg.setAttribute("src",episode.image.medium)
+    episodeDescription.innerHTML=episode.summary;
+    optionEl.innerHTML=`S0${episode.season} E${epiNo} - ${episode.name}`;
+    optionEl.value=episode.id;
+
+    //Append the child element
+    divEl.appendChild(episodeBtn);
+    divEl.appendChild(episodeImg);
+    divEl.appendChild(episodeDescription);
+    rootElem.appendChild(divEl);
+    selectEl.appendChild(optionEl)
+  })
 }
 
-window.onload = setup;
+// Create Live Search 
+const searchInput=document.getElementById("searchInput");
```

Comment:
Currently, we are working with a local file as a source of our data, so we are not bombarding API yet ;)  
But before we are going to connect to the API, please consider using debounce() function. 
This solution allows us to put on hold sending the get request and wait, for example, 1.5 seconds after the user stopped typing the search word. 
https://www.educative.io/answers/how-to-use-the-debounce-function-in-javascript
https://www.youtube.com/watch?v=yBFHwJgqLD4